Client/src/test/functional/com/agilepro/client/helpers/TFBackLogHelper.java

```diff
@@ -0,0 +1,218 @@
+		 **/
+		CustomerPricePlanModel customerPricePlanModel = new CustomerPricePlanModel();
+		customerPricePlanModel.setName("Test12");
+		customerPricePlanModel.setPaymentCycle(PaymentCycle.Daily);
+		CustomerPricePlanExpression ex = new CustomerPricePlanExpression();
+		ex.setExpression("3+2");
+		ex.setName("test11");
+		ex.setLabel("pay");
+		List<CustomerPricePlanExpression> listExp = new ArrayList<CustomerPricePlanExpression>();
+		listExp.add(ex);
+		customerPricePlanModel.setExpressions(listExp);
+		long idOfPricePlan = pricePlanHelper.save(clientContext, customerPricePlanModel);
+		Assert.assertTrue(idOfPricePlan > 0);
+
+		/**
+		 * customer.
+		 **/
+		CustomerModel model = new CustomerModel("Test1", emailId, phoneNumber, null, null, null, null, new Date(), password, password, "path1", null);
+		model.setCustomerPricePlanId(idOfPricePlan);
+		model.setDueAmount(dueAmount);
+		customerId = customerHelper.save(clientContext, model);
+		Assert.assertTrue(idOfPricePlan > 0);
+		Assert.assertTrue(customerId > 0);
+		clientCurrentSession = super.newClientContext(emailId, password, customerId);
+		// camp type
+
+		clientControllerFactory = new ClientControllerFactory(clientCurrentSession);
+
+	}
+
+	/**
+	 * Gets the backlog.
+	 *
+	 * @param backlogId
+	 *            the backlog id
+	 * @return the backlog
+	 */
+	private BackLogModel getBacklog(long backlogId)
+	{
+		BasicReadResponse<BackLogModel> response = backlogcontroller.read(backlogId);
+		return response.getModel();
+	}
+
+	/**
+	 * Test save.
+	 */
+	@Test
+	public void testSave()
+	{
+		clientControllerFactory = new ClientControllerFactory(clientCurrentSession);
+
+		backlogcontroller = (IBackLogController) super.clientControllerFactory.getController(IBackLogController.class);
+
+		BackLogModel model = new BackLogModel();
+		model.setTitle("studentGrade");
+		model.setDescription("hello");
+		model.setEstimate(23);
+		BasicSaveResponse response2 = backlogcontroller.save(model);
+
+		Assert.assertEquals(response2.getCode(), IWebUtilsCommonConstants.RESPONSE_CODE_SUCCESS);
+		// validate save
+		getBacklog(response2.getId());
+	}
+
+	/**
+	 * Test update.
+	 */
+	@Test
+	public void testUpdate()
+	{
+		clientControllerFactory = new ClientControllerFactory(clientCurrentSession);
+
+		backlogcontroller = (IBackLogController) super.clientControllerFactory.getController(IBackLogController.class);
+
+		BackLogModel model = new BackLogModel();
+		model.setTitle("studentGrade");
+		model.setDescription("hello");
+		model.setEstimate(23);
+
+		// validate save
+		BasicSaveResponse response3 = backlogcontroller.save(model);
+
+		Assert.assertEquals(response3.getCode(), IWebUtilsCommonConstants.RESPONSE_CODE_SUCCESS);
+
+		// validate save
+		getBacklog(response3.getId());
+
+		BackLogModel model2 = new BackLogModel();
+		model2.setTitle("studentGraxxde");
+		model2.setDescription("hello");
+		model2.setEstimate(23);
+		model2.setId(response3.getId());
+		backlogcontroller.update(model2);

```

Comment:
Where is the validation post update. This is just invoking update, no validations are present.